### PR TITLE
test: registry unit tests

### PR DESCRIPTION
**Pull request by 0xOneTony**

### CODE REVIEW

File: test/unit/core/RegistryUnit.t.sol [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    function test__generateProfileIdShouldReturnTheKeccak256OfEncodedNonceAndOwnerAddress(
+        uint256 _nonce,
+        address _owner
+    ) external {
+        // it should return the keccak256 of encoded nonce and owner address
+        assertEq(keccak256(abi.encodePacked(_nonce, _owner)), registry.call__generateProfileId(_nonce, _owner));
+    }
+
+    function test__isOwnerOfProfileWhenProvidedAddressIsOwnerOfProfile(bytes32 _profileId, address _owner) external {
+        vm.skip(true);
```
Comment (0xAustrian) [inline test/unit/core/RegistryUnit.t.sol:306]: Is something wrong with the test?

File: test/unit/core/RegistryUnit.tree [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    └── when called
+        └── it should call _revokeRole for members
+
+Registry::_checkOnlyProfileOwner
+├── when caller is not profile owner
+│   └── it should revert
+└── when caller is profile owner
+    └── it should not revert
+
+Registry::_generateAnchor
```
Comment (0xAustrian) [inline test/unit/core/RegistryUnit.tree:72]: why empty?

**Review by 0xAustrian — COMMENTED** [review]
Looks good, I see a few missing tests or skipped, why is that?

File: test/unit/core/RegistryUnit.t.sol (hunk 1 above)
Comment (0xOneTony) [reply test/unit/core/RegistryUnit.t.sol:306]: something wrong with stdstorage couldnt make it work for now

File: test/unit/core/RegistryUnit.tree (hunk 2 above)
Comment (0xOneTony) [reply test/unit/core/RegistryUnit.tree:72]: oups nice catch will add them now

File: test/unit/core/RegistryUnit.t.sol (hunk 1 above)
Comment (0xAustrian) [reply test/unit/core/RegistryUnit.t.sol:306]: You can add a TODO so we can fix it later

**Review by 0xAustrian — APPROVED** [review]

**Review by thelostone-mc — APPROVED** [review]

**Review by ilpepepig — APPROVED** [review]

**Pull request merged**